### PR TITLE
Add converter from internal span to Jaeger span

### PR DESCRIPTION
**Pull request by black-adder**

### CODE REVIEW

File: jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import java.util.Map;
+
+public class JaegerThriftSpanConverter {
+
+  public static com.uber.jaeger.thriftjava.Span convertSpan(Span span) {
+    SpanContext context = span.context();
+
+    return new com.uber.jaeger.thriftjava.Span(
+            context.getTraceID(),
+            0,
```
Comment (yurishkuro) [inline jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java]: what is this 0?  can you add a comment

File: jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      tag.setVBool((Boolean) tagValue);
+    } else {
+      tag = buildStringTag(tag, tagValue);
+    }
+    return tag;
+  }
+
+  protected static Tag buildStringTag(Tag tag, Object tagValue) {
+    tag.setVType(TagType.STRING);
+    String stringTagValue = tagValue.toString();
```
Comment (yurishkuro) [inline jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java]: s/toString/String.valueOf()/, it protects against null

File: jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    tag.setVStr(truncateString(stringTagValue));
+    return tag;
+  }
+
+  protected static String truncateString(String s) {
+    // TODO rename to MAX_TAG_LENGTH
```
Comment (yurishkuro) [inline jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java]: let's do it now

File: jaeger-core/src/test/java/com/uber/jaeger/reporters/protocols/InMemorySpanServerHandler.java [hunk 4]
```diff
@@ -38,6 +39,14 @@ public void emitZipkinBatch(List<Span> spans) throws TException {
     }
   }
 
+  @Override
+  public void emitBatch(Batch batch) throws TException {
+    synchronized (this) {
+      // TODO convert spans
+      //this.spans = batch.spans;
```
Comment (yurishkuro) [inline jaeger-core/src/test/java/com/uber/jaeger/reporters/protocols/InMemorySpanServerHandler.java]: ?

File: jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.uber.jaeger.reporters.protocols;
+
+import com.uber.jaeger.*;
```
Comment (vprithvi) [inline jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java]: Import only the classes you need. 
(Not sure why this wasn't caught by our build)

File: jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        jTags.add(buildTag(tagKey, tagValue));
+      }
+    }
+    return jTags;
+  }
+
+  protected static Tag buildTag(String tagKey, Object tagValue) {
+    Tag tag = new Tag();
+    tag.setKey(tagKey);
+    if (tagValue instanceof Number) {
```
Comment (vprithvi) [inline jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java]: Why is this check needed? 
It seems that the handling of `AtomicInteger`, etc is to simply add it to a string tag, which happens as the default fallback

File: jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java (hunk 2 above)
Comment (vprithvi) [inline jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java]: +1

File: jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    } else if (tagValue instanceof Boolean) {
+      tag.setVType(TagType.BOOL);
+      tag.setVBool((Boolean) tagValue);
+    } else {
+      tag = buildStringTag(tag, tagValue);
+    }
+    return tag;
+  }
+
+  protected static Tag buildStringTag(Tag tag, Object tagValue) {
```
Comment (vprithvi) [inline jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java]: Do we need to take in the `Tag` object?

**Pull request merged**